src/plugins/auto/executable_network.cpp

```diff
@@ -29,24 +29,28 @@ using namespace InferenceEngine;
 
 namespace {
 std::string GetNetworkPrecision(const InferenceEngine::CNNNetwork &network) {
-    auto nGraphFunc = network.getFunction();
-    bool isINTModel = ngraph::op::util::has_op_with_type<ngraph::op::FakeQuantize>(nGraphFunc);
-    if (isINTModel) {
-        return METRIC_VALUE(INT8);
-    }
-    for (auto & node : nGraphFunc->get_ordered_ops()) {
-        if (std::dynamic_pointer_cast<ngraph::opset1::Convolution>(node) ||
-            std::dynamic_pointer_cast<ngraph::opset1::GroupConvolution>(node) ||
-            std::dynamic_pointer_cast<ngraph::opset1::GroupConvolutionBackpropData>(node) ||
-            std::dynamic_pointer_cast<ngraph::opset1::ConvolutionBackpropData>(node)) {
-            auto layerType = node->input(1).get_element_type().get_type_name();
-            if (layerType == "f32")
-                return METRIC_VALUE(FP32);
-            if (layerType == "f16")
-                return METRIC_VALUE(FP16);
+    try {
```

Comment:
if  we can use 'if(modelpath.empty)',  assert network is not null and not need try catch

Reply:
if network is not available, shall we get precision from modelpath?

Reply:
network should be available, because modelpath is empty, you will read it from modelpath in this PR

Reply:
fixed